### PR TITLE
🌱 Bump versions for v1.2

### LINKED ISSUE #6020: Prepare main branch for v1.2 development
As we're now starting work on v1.2 let's prepare the main branch for v1.2 development.

This issue is mostly about changing all occurrences of the CAPI v1.1.x  version to v1.2.x

As far as I'm aware we have to do the following: 
* Add new release to top-level metadata.yaml and test/e2e/data/shared/v1beta1/metadata.yaml
* Update providers in docker.yaml. Roughly:
    * create a new entry for each provider for v1.1.0 (similar to the existing ones for v1.0.1)
    * rename v1.1.99 to v1.2.99
* Update create-local-repository.py: v1.1.99 => v1.2.99
* Update tools/tilt-prepare/main.go: v1.1.99 => v1.2.99
* Make sure CI is green (on the PR: `/test pull-cluster-api-e2e-full-main` and `/test pull-cluster-api-e2e-workload-upgrade-1-23-latest-main`)

/kind feature

### PR DESCRIPTION
**Pull request by alexander-demichev**
**What this PR does / why we need it**:

Add v1.2 to metadata.yaml since there is already release-1.1 branch.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6020

### CODE REVIEW

File: metadata.yaml [hunk 1]
```diff
@@ -6,6 +6,9 @@
 apiVersion: clusterctl.cluster.x-k8s.io/v1alpha3
 kind: Metadata
 releaseSeries:
+  - major: 1
+    minor: 2
+    contract: v1beta1
```
Comment (fabriziopandini) [inline metadata.yaml:11]: @sbueringer  do we need to have the same change in test/e2e/data/shared/v1beta1/metadata.yaml too?

Comment (sbueringer) [reply metadata.yaml:11]: Yes.

We need a few more changes in test data, but I think some of them would be better done after the v1.1 release.

But I think it's a good idea to keep the metadata.yaml in sync in this PR.

Comment (alexander-demichev) [reply metadata.yaml:11]: I've just updated metadata in e2e, can you take a look?

Comment (sbueringer) [reply metadata.yaml:11]: Looks perfect!

Comment (alexander-demichev) [reply metadata.yaml:11]: yes, looks like CI won't be happy until 1.1 is released. Let's hold the PR until next week.

**Comment by alexander-demichev** [comment]
/hold
we should wait for v.1.1 release

File: metadata.yaml (hunk 1 above)
Comment (fabriziopandini) [reply metadata.yaml:11]: +1 make a bulk of all the changes above

Comment (sbueringer) [reply metadata.yaml:11]: I'll open an issue hopeful until EOW which captures/summarizes what we want to do. @alexander-demichev I'll mention you on it.

**Comment by vincepri** [comment]
We should be able to merge this now that we have the 1.1 release branch

**Comment by sbueringer** [comment]
> We should be able to merge this now that we have the 1.1 release branch

In theory yes, in practice this change breaks CI when we keep the test metadata.yaml in sync.

This PR will be extended to implemented this entire issue: https://github.com/kubernetes-sigs/cluster-api/issues/6020 after the v1.1.0 release. 

We could implement it partially already, but it's probably not worth having the repo in an inconsistent state for a few days.

**Comment by alexander-demichev** [comment]
@fabriziopandini @sbueringer I updated this PR.

**Comment by sbueringer** [comment]
@alexander-demichev Can you please update the PR description to contain `Fixes #6020`? Thx!

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**